### PR TITLE
Add prelim support for MPICH

### PR DESCRIPTION
**Pull request by rhc54**
Harvest some envars, set a few on fork

Signed-off-by: Ralph Castain <rhc@pmix.org>

### CODE REVIEW

**Comment by rhc54** [comment]
@raffenet @zhenggb72 @wesbland

I have made a first pass at adding MPICH support into PMIx - just harvesting/forwarding envars and setting some for each process. I'm not sure entirely what you folks support. I based this first cut on what I saw Hydra doing from your latest 4.0 alpha tarball.

Can you please take a look and tell me what I got wrong and/or what might be missing? I'm working next on Hydra-like support in PRRTE and should have something for you to review there soon. These will be too late for the next release (which is going to happen in the next week), but we can include them in a subsequent release in about a month.

**Comment by raffenet** [comment]
Just acknowledging that I have seen this and will dig into it when I have some time. Thanks!

**Comment by rhc54** [comment]
bot:ibm:prrte:retest

File: src/mca/pmdl/mpich/pmdl_mpich_component.c [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    },
+    .include = NULL,
+    .exclude = NULL
+};
+
+static pmix_status_t component_register(void)
+{
+    pmix_mca_base_component_t *component = &mca_pmdl_mpich_component.super.base;
+
+    mca_pmdl_mpich_component.incparms = "MPIR_*";
```
Comment (zhenggb72) [inline src/mca/pmdl/mpich/pmdl_mpich_component.c:68]: I think it is safe to use "MPIR_CVAR" for all MPICH env vars.

Comment (rhc54) [reply src/mca/pmdl/mpich/pmdl_mpich_component.c:68]: Good point. Question: Hydra has options about forwarding envars. It wasn't quite clear to me if those options apply to these envars, or were aimed at general envars other than the MPICH-specific ones. Any insights would be appreciated.

**Pull request merged**